frontend/src/pages/ProfilePage/ProfilePage.jsx

```diff
@@ -117,7 +123,7 @@ const ProfilePage = () => {
       const response = await axiosInstance.put(`/update-email`, { newEmail });
       console.log("Response from API:", response);
 
-      if (response.data) {
+      if (response.data&&validateEmail(newEmail)) {
```

Comment:
I believe it might be helpful to validate the email before making the PUT API call. This way, we can avoid saving invalid emails in the database in the first place.

It would also improve the user experience by ensuring they are notified promptly if their email is invalid. This could help streamline the process and prevent potential issues down the line.